chore: QA WG's new OKRs (#287)
electron/governance: wg-releases/wg-qa/README.md

## Current Objective and Key Results

**Objective 1:**

> Early detection of Electron issues

Reviewer: So using the guidance from my general comment I think we could make the objectives more concrete. For example (and this is just something I thought for a minute and definitely not what I'm proposing as the final objective):

```suggestion
All compatibility issues between new Electron versions and current apps are detected before releasing a new major version
```
 
This is highly inspirational, and it will take quite a few different things and time to achieve (or even get close to that), so we can keep adding key results to the list as we make progress.

Author: Major version issues* between new Electron versions and current apps are detected before releasing a new major version

*Includes: chromium, node, & electron issues

**Key Results:**

> * Report alpha/beta electron issues on latest 2 release versions and beta version (as a team)

Reviewer: What is the output of this KR?  Is it going to be a document or a page with all the versions for the latest 2 released and all the issues for each one?
How do we consider the KR complete? There can always be a new bug we didn't find before, right?

Reviewer 2: This is likely an evergreen KR - as a team, we report on and discuss issues in the monthly team syncs as well as the weekly reports in channel to give others a heads up, which has worked well several times.
There can definitely be a bug we didn't find before. I'd like to see what others think of the output/completion of this KR.

Reviewer: Because the team already has a workflow in place, maybe this KR should be replaced by another one? Or maybe it could be a new way of reporting that we believe will be more effective?

My issue is that with the current framing and what you've told me, what is going on is already a "given". We are not making any progress and just maintaining the status quo.

Does it make sense?

Author: ~~* Report alpha/beta electron issues on latest 2 release versions and beta version (as a team)~~
* At least one app to integrate Sentinal

Reviewer 3: hey folks! i'm excited yall want to try out my ongoing work, but i definitely want to make sure we're aligned on requirements and restraints of the project and what an app would need to be able to report back to the service in order to get the most out of it, since it is in alpha and not in a state where there's available documentation for said constraints.

What's the timeframe for this being merged? Perhaps i should visit the next QA meeting?

Author: @codebytere our next meeting is 7/6. We were going to merge like tomorrow'ish

> * Collaborate on and deliver a list of api tests (via fiddle?) that can be automated

Reviewer: Does "collaborate" here mean a team effort or are we going to work with other teams/companies?
Do we have an idea of how long that list should be? Is it going to be a percentage of the API surface (e.g. 90%) or a raw number (maybe based on what we know are the most used ones)?

Reviewer 2: Collaborate here means a team effort of the WG-QA.

To me, it makes more sense to base the list on the most used APIs.

Reviewer: It also makes more sense to me 😊
Do we know how to get that information?

Author: * Audit current automated API tests and document results

Note: would be interesting to get some telemetry on most popular APIs

**Objective 2:**

> Ease the pain of QA or QAless teams moving to Electron

Reviewer: If I've understood this correctly, we want QA teams from companies say "we like Electron because their QA guidance is 💯", is that right?

"ease the pain" might be a bit negative. What about making it a bit more inspirational? Maybe something around these lines:

```suggestion
Electron has the best QA documentation of any platform
```

It is still not super concrete, but we could run some surveys and such to make sure the satisfaction is high.

**Key Results:**

> * Document each current members QA structure and process

Reviewer: I don't think I know what this means. Do you mind explaining it to me? 🙏

Reviewer 2: Team composition, structure, and processes for approaching Electron testing vary significantly from company to company... Some have dedicated QA teams that focus on testing Electron only (i.e. Slack) while others have no QA support at all or limited resources that split their time between integrating with Electron and verifying their own App features.

The goal of this KR is to document each member's structure and process, share and learn from each company's experience.

Reviewer: And by member's we are talking about each person working on a project that is part of wg-qa, right?

I ❤ this!

Author: * Document each app's QA structure and process

> * Build a list of tooling needs for accurate and fast Electron application testing

Reviewer: 😍
Is the goal to have it published in the website by the end of the period?

Reviewer 2: Yes, I believe so

Reviewer: Then I think we should make it clear in the KR.
I had one problem in the past where we weren't clear enough on the output and the team had different interpretations (and obviously our manager picked the one that meant way more work).

Author: * Publish on electronjs.org a list of tooling needs for accurate and fast Electron application testing
